projects/igniteui-angular/src/lib/tabs/tabs.component.ts

```diff
@@ -422,6 +358,105 @@ export class IgxTabsComponent implements IgxTabsBase, AfterViewInit, OnDestroy {
             }
         }
     }
+
+    /**
+     * @hidden
+     */
+    public performSelectionChange(newTab: IgxTabItemBase): void {
+        const oldTab = this.selectedTabItem;
+        if (oldTab) {
+            this.performDeselection(oldTab);
+        }
+        if (newTab) {
+            this.performSelection(newTab);
+        } else {
+            // if there is no new selected tab hide the visibility indicator
+            this.hideIndicator();
+        }
+    }
+
+    private performDeselection(oldTab: IgxTabItemBase): void {
+        oldTab._isSelected = false;
+        const oldTabRelatedGroup = this.groups.toArray()[oldTab.index];
+        if (oldTabRelatedGroup) {
+            oldTabRelatedGroup._isSelected = false;
+        }
+        this._selectedIndex = -1;
+        this.onTabItemDeselected.emit({ tab: oldTab, group: oldTabRelatedGroup });
+    }
+
+    private performSelection(newTab: IgxTabItemBase): void {
+        newTab._isSelected = true;
+        newTab.tabindex = 0;
```

Comment:
`tabindex` should be set to -1 on deselection.

Reply:
Addressed.